core/modules/update/update.install

```diff
@@ -239,6 +239,31 @@ function update_update_1004() {
   $config->save();
 }
 
+/**
+ * Check installer.settings.json and fix it or remove it if needed.
+ */
+function update_update_1005() {
+  $config = config('installer.settings');
+  if (!$config->isNew()) {
```

Comment:
This is TRUE if the settings file doesnt exist, right? Does this block need to run if the file doesnt exist?

Reply:
The idea here is that the fixing or removal of the config file must happen only if the file actually exists, and if it's faulty (e.g. if it's there even when `installer` is disabled, or if the key `installer_server` doesn't exist in the file).

So, this `if` statement checks the config file. If the file is new, that means that the file doesn't exist. If this is the case, we don't need to do any fixing or removing. 

Side note: a different PR, already committed and released as part of 1.22.2, already fixed the code that created the faulty and unnecessary `installer.settings.json` file in the first place.

Reply:
> If the file is new, that means that the file doesn't exist

That sounds so totally backwards! 😄 Sounds like we should have a `Config::file_exists()` or something. 

In either case, maybe a brief code comment for the rest of us please?

Reply:
> That sounds so totally backwards!

Yup 😄 My wording didn't help. Rephrase: "if the config object returns TRUE for `isNew()` that means that it has not been saved, and the file doesn't exist." Makes more sense?  

I'll add better documentation later.

Reply:
@docwilmot, I added more explanations. Please check to see if they make sense now.